core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -71,64 +78,99 @@ object ConsumerGroupCommand {
       }
     } catch {
       case e: Throwable =>
-        println("Error while executing consumer group command " + e.getMessage)
-        println(Utils.stackTrace(e))
+        consumerGroupOutputWriter.printError("Executing consumer group command failed due to " + e.getMessage, Some(e))
     } finally {
       consumerGroupService.close()
     }
   }
 
-  sealed trait ConsumerGroupService {
+  case class PartitionAssignmentState(group: String, topic: Option[String], partition: Option[Int], offset: Option[Long],
+                                      lag: Option[Long], memberId: Option[String], memberHost: Option[String],
+                                      clientId: Option[String], logEndOffset: Option[Long])
 
-    def list(): Unit
+  class ConsumerGroupOutputWriter extends Logging {
 
-    def describe() {
-      describeGroup(opts.options.valueOf(opts.groupOpt))
+    def printError(msg: String, e: Option[Throwable] = None): Unit = {
+      println("Error: " + msg)
+      if (e.isDefined)
+        e.map(debug("Exception in consumer group command", _))
     }
 
-    def close(): Unit
+    def printMessage(msg: String): Unit = println(msg)
 
-    protected def opts: ConsumerGroupCommandOptions
+    def printAssignment(groupAssignment: Array[PartitionAssignmentState]): Unit = {
+      println("%-30s %-30s %-10s %-15s %-15s %-10s %-50s %-30s %s".format("GROUP", "TOPIC", "PARTITION", "CURRENT-OFFSET", "LOG-END-OFFSET", "LAG", "MEMBER-ID", "MEMBER-HOST", "CLIENT-ID"))
+      groupAssignment.foreach { memberAssignment =>
+        println("%-30s %-30s %-10s %-15s %-15s %-10s %-50s %-30s %s".format(
+          memberAssignment.group, memberAssignment.topic.getOrElse(""), memberAssignment.partition.getOrElse(""),
+          memberAssignment.offset.getOrElse(""), memberAssignment.logEndOffset.getOrElse(""), memberAssignment.lag.getOrElse(""),
+          memberAssignment.memberId.getOrElse("none"), memberAssignment.memberHost.getOrElse(""), memberAssignment.clientId.getOrElse("")))
+      }
+    }
+  }
 
-    protected def getLogEndOffset(topic: String, partition: Int): LogEndOffsetResult
+  sealed trait ConsumerGroupService {
 
-    protected def describeGroup(group: String): Unit
+    def list(): List[String]
```

Comment:
This is looking promising. Maybe we could rename `list` to `listGroups` and `describe` to `describeAssignment`?